test/integration/fact_check_email_test.rb

```diff
@@ -1,58 +1,68 @@
 require "legacy_integration_test_helper"
+require "gmail_test_helper"
 
 class FactCheckEmailTest < LegacyIntegrationTest
   def fact_check_mail_for(edition, attrs = {})
-    message = Mail.new do
-      from    attrs.fetch(:from,    "foo@example.com")
+    # Still have to construct the basic Mail object so we can produce a RFC-2822 raw value for the Gmail message
+    base_mail = Mail.new do
+      from    attrs.fetch(:from, "foo@example.com")
       to      attrs.fetch(:to,      edition && edition.fact_check_email_address)
       cc      attrs.fetch(:cc,      nil)
       bcc     attrs.fetch(:bcc,     nil)
       subject attrs.fetch(:subject, "This is a fact check response [#{edition.present? ? edition.id : ''}]")
       body    attrs.fetch(:body,    "I like it. Good work!")
     end
 
-    # The Mail.all(:delete_after_find => true) call in FactCheckEmailHandler will set this
-    # on all messages before yielding them
-    message.mark_for_delete = true
-    message
+    build_gmail_message(
+      {
+        To: attrs.fetch(:to, edition && edition.fact_check_email_address),
+        From: attrs.fetch(:from, "foo@example.com"),
+        CC: attrs.fetch(:cc, nil),
+        BCC: attrs.fetch(:bcc, nil),
+        Subject: attrs.fetch(:subject, "This is a fact check response [#{edition.present? ? edition.id : ''}]"),
+      },
+      attrs.fetch(:body, "I like it. Good work!"),
+      base_mail.to_s,
+    )
+  end
+
+  def handler(messages)
+    handler = FactCheckEmailHandler.new(fact_check_config, stub.stubs(:set))
```

Comment:
stub.stubs is a bit confusing, maybe we could define it separately?

Reply:
Despite this being marked as new code, this is 1:1 what currently exists in our test framework. I just moved its location https://github.com/alphagov/publisher/blob/5325b16b97bf180609711494c93f4a09f36c2310/test/integration/fact_check_email_test.rb#L28 

I was trying not to rock the boat too much - this stub was put in place by Richard when putting in the prometheus metrics and is purely there to "eat" that call as we're not considering prometheus here.

Reply:
Oh yeah, sorry for the bad variable name 😓  - I think let's leave it if it was already there. I might have a look and see if it can be made better in a separate PR.